webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTable.js

```diff
@@ -78,40 +77,33 @@ class SubscriptionsTable extends Component {
     const allSubscriptionResults = this.props.subscriptions.results;
 
     const checkAllRowsSelected = () =>
-      allSubscriptionResults.length === this.state.selectedRows.length;
+      allSubscriptionResults.length === this.props.selectedRows.length;
 
     const updateDeleteButton = () => {
-      this.props.toggleDeleteButton(this.state.selectedRows.length > 0);
+      this.props.toggleDeleteButton(this.props.selectedRows.length > 0);
     };
     return ({
       allRowsSelected: () => checkAllRowsSelected(),
       selectAllRows: () => {
         if (checkAllRowsSelected()) {
-          this.setState(
-            { selectedRows: [] },
-            updateDeleteButton,
-          );
+          this.props.onSelectedRowsChange([]);
+          this.setState(updateDeleteButton);
```

Comment:
It looks like this is passing the `updateDeleteButton` function to `this.setState`.  Since `updateDeleteButton` just invokes `this.props.toggleDeleteButton()`, and doesn't return an object describing the new state, this `setState` call doesn't make sense to me.  Can probably change it to
```js
if (checkAllRowsSelected()) {
  this.props.onSelectedRowsChange([]);
  this.props.toggleDeleteButton(false);  // will always be false because we just deselected all rows
  ...
```

Reply:
Explanation of ^^

In the previous code, `updateDeleteButton` was the 2nd argument to `this.setState`, meaning it's a callback that React runs after the state update is completed.  But in the new code, it's the 1st argument, meaning it's expected to be an "[updater](https://reactjs.org/docs/react-component.html#setstate)" function that returns the new state.